lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -96,6 +96,7 @@ class _Visitor extends SimpleAstVisitor<void> {
           return;
         }
       } else if (parent is MethodInvocation) {
+        if (expression is CascadeExpression) return;
```

Comment:
This condition can be moved to the root of the method, in this case, there is a second check (in line 150 now), which should be removed.

Another solution is to use the implementation of the other PR.

Reply:
Ah!  This check here is scoped to just the null aware and simple identifier cases so I'm not sure it's identical to the check in 150 but I'm sure you're right that this could be tidied up.  (Lots of special cases here...)

Feel free to follow-up w/ any refinements?  It'd be great to flesh out the tests too.  (FYI @oprypin who may have more cases to consider.)